readthedocs/search/parsers.py

```diff
@@ -140,21 +140,68 @@ def _parse_sections(self, title, body):
         except Exception as e:
             log.info('Unable to index section', section=str(e))
 
+        # All terms in dls are treated as sections.
+        # After indexing them, we remove them from the body such that their contents aren't
+        # repeated in the following indexing
+        dls = body.css("dl")
+        for dl in dls:
+            # Select all dts with id defined
+            dts = dl.css('dt[id]:not([id=""])')
+
+            # https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dt
+            # multiple <dt> elements in a row indicate several terms that are
+            # all defined by the immediate next <dd> element.
+            for dt in dts:
+                try:
+                    title, _id = self._parse_dt(dt)
+                    # https://developer.mozilla.org/en-US/docs/Web/CSS/General_sibling_combinator
+                    dd = dt.css_first("dt ~ dd")
+                    # We only index a dt with an id attribute and an accompanying dd
+                    if not dd or not _id:
+                        continue
+
+                    # The content of the <dt> section is the content of the accompanying <dd>
+                    content = dd.text()
+                    yield {
+                        "id": _id,
+                        "title": title,
+                        "content": content,
+                    }
+                except Exception:
+                    log.info("Unable to index dt section.", exc_info=True)
+
+            # Remove the <dl> node
+            # There isn't a clear indication if this behavior is DFS or BFS
+            dl.decompose()
+
+        # This is pattern is intended for a future generic scenario
+        # We could include the <dl> pattern in _is_section etc. instead of having the above
+        # <dl> block
         # Index content from h1 to h6 headers.
-        for head_level in range(1, 7):
-            tags = body.css(f'h{head_level}')
-            for tag in tags:
+        outer_sections = [body.css(f"h{h}") for h in range(1, 7)]
+        for section in outer_sections:
```

Comment:
This is still iterating the nodes twice, but even if we use a generator expression, I'd prefer if we don't add code that we don't know yet if we are going to need.